File: src/main/java/com/rokucraft/rokubot/entities/Tag.java
package com.rokucraft.rokubot.entities;

import net.dv8tion.jda.api.utils.messages.MessageCreateData;
import org.checkerframework.checker.nullness.qual.Nullable;
import org.spongepowered.configurate.objectmapping.ConfigSerializable;
import org.spongepowered.configurate.objectmapping.meta.Required;

@ConfigSerializable
public record Tag(
        @Required String name,
        @Nullable String description,
        @Required MessageCreateData message
) {}

File: src/main/kotlin/com/rokucraft/rokubot/entities/Tag.kt
package com.rokucraft.rokubot.entities

import net.dv8tion.jda.api.utils.messages.MessageCreateData
import org.spongepowered.configurate.objectmapping.ConfigSerializable
import org.spongepowered.configurate.objectmapping.meta.Required

@ConfigSerializable
@JvmRecord
data class Tag(
    @Required val name: String,
    val description: String?,
    @Required val message: MessageCreateData
)
